media-capabilities/index.js

```diff
@@ -16,22 +16,22 @@
 
 const supported = typeof window !== 'undefined' && 'mediaCapabilities' in navigator;
 
-const useMediaCapabilities = (mediaConfig, initialMediaCapabilities = {}) => {
-  let mediaCapabilities = {
+const useMediaCapabilitiesDecodingInfo = (mediaDecodingConfig, initialMediaCapabilitiesInfo = {}) => {
+  let mediaCapabilitiesInfo = {
     supported,
-    hasMediaConfig: !!mediaConfig
+    hasMediaDecodingConfig: !!mediaDecodingConfig
   };
 
-  mediaCapabilities = (mediaCapabilities.supported && mediaCapabilities.hasMediaConfig)
-    ? navigator.mediaCapabilities.decodingInfo(mediaConfig)
+  mediaCapabilitiesInfo = (mediaCapabilitiesInfo.supported && mediaCapabilitiesInfo.hasMediaDecodingConfig)
+    ? navigator.mediaCapabilities.decodingInfo(mediaDecodingConfig)
```

Comment:
This still doesn't take into account the fact this returns a promise.

Reply:
WDYT about using an API like,

```js
const { supported, decodingInfo, encodingInfo } = useMediaCapabilities(initialState)
```

either that or we need to wrap the async calls and expose them to show the state of the calls.

Reply:
@wingleung cc @addyosmani 

What are your thoughts?
I agree with @stramel's API suggestion. I think it looks more descriptive rather than just `mediaCapabilitiesInfo`.
For now, I think it could be like so.
```
const { supported, smooth, powerEfficient } = useMediaCapabilities(initialState);
```